homeassistant/components/weather/__init__.py

```diff
@@ -1178,6 +1211,56 @@ async def async_get_forecasts_service(
     }
 
 
+async def async_get_forecast_attr_service(
+    weather: WeatherEntity, service_call: ServiceCall
+) -> ServiceResponse:
+    """Get weather forecast values for specified attribute."""
+    forecast_type = service_call.data["type"]
+    forecast_attribute = service_call.data["attribute"]
+    forecast_limit = service_call.data.get("limit")
+
+    supported_features = weather.supported_features or 0
+    if forecast_type == "daily":
+        if (supported_features & WeatherEntityFeature.FORECAST_DAILY) == 0:
+            raise_unsupported_forecast(weather.entity_id, forecast_type)
+        native_forecast_list = await weather.async_forecast_daily()
+    elif forecast_type == "hourly":
+        if (supported_features & WeatherEntityFeature.FORECAST_HOURLY) == 0:
+            raise_unsupported_forecast(weather.entity_id, forecast_type)
+        native_forecast_list = await weather.async_forecast_hourly()
+    else:
+        if (supported_features & WeatherEntityFeature.FORECAST_TWICE_DAILY) == 0:
+            raise_unsupported_forecast(weather.entity_id, forecast_type)
+        native_forecast_list = await weather.async_forecast_twice_daily()
+    if native_forecast_list is None:
+        converted_forecast_list = []
+    else:
+        # pylint: disable-next=protected-access
+        converted_forecast_list = weather._convert_forecast(native_forecast_list)
+
+    if forecast_limit is not None:
+        # Should not be greater than the length of the list
+        if forecast_limit > len(converted_forecast_list):
+            # Cut list length according to forecast_limit_hours
+            converted_forecast_list = converted_forecast_list[:forecast_limit]
+
+    # Check if key forecast_attribute is in list
+    if forecast_attribute not in converted_forecast_list[0]:  # type: ignore[operator]
+        raise HomeAssistantError(
+            f"Forecast attribute '{forecast_attribute}' does not exist in forecast"
+        )
```

Comment:
```suggestion
    if forecast_attribute not in converted_forecast_list[0]:  # type: ignore[operator]
        raise ServiceValidationError(
            f"Forecast attribute '{forecast_attribute}' does not exist in forecast"
        )
```
Would be good if we add translations for the error also https://developers.home-assistant.io/docs/internationalization/core#exceptions